rust/signed_doc/src/lib.rs

```diff
@@ -64,7 +65,55 @@ impl From<InnerCatalystSignedDocument> for CatalystSignedDocument {
     }
 }
 
+/// Catalyst Signed Document Error
+#[derive(Debug)]
+pub struct CatalystSignedDocError {
+    /// List of errors during processing.
+    report: ProblemReport,
+    /// Actual error.
+    error: anyhow::Error,
+}
+
+impl fmt::Display for CatalystSignedDocError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let report_json = serde_json::to_string(&self.report)
+            .unwrap_or_else(|_| String::from("Failed to serialize ProblemReport"));
+
+        write!(
+            fmt,
+            "CatalystSignedDocError {{ error: {}, report: {} }}",
+            self.error, report_json
```

Comment:
I don't know if it is intended, but `anyhow::Error` has very different implementations of `Display` and `Debug` traits. The former one prints only the error itself, while the latter one prints the whole error chain that can contain more useful information. Unless we specifically don't want to be too verbose, I would prefer to use the debug representation.

Reply:
I think as we are implementing `Display` trait here, we dont want to be too verbose and provide `Debug` information about the error at this place. If it will be needed more verbose output, it could be used `{:?}` debug formatting for the `CatalystSignedDocError` object itself.
So we will not make a decision how to print at this place actually, and the user will have a possibility to choose.